command/operator_diagnose.go

```diff
@@ -257,22 +253,24 @@ func (c *OperatorDiagnoseCommand) offlineDiagnostics(ctx context.Context) error
 
 		// Attempt to use storage backend
 		if !c.skipEndEnd {
-			err = diagnose.StorageEndToEndLatencyCheck(ctx, b)
+			err = diagnose.StorageEndToEndLatencyCheck(ctx, *backend)
 			if err != nil {
 				return err
 			}
 		}
 
 		return nil
 	}); err != nil {
-		return err
+		diagnose.Error(ctx, err)
```

Comment:
I'm not sure you'll need these errors, as they'd be caught by Test as well.  I think you can just drop the if entirely

Reply:
Ooh, that would be a lot nicer. I'll give that a try!